packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx

```diff
@@ -316,20 +316,23 @@ export class NavBase extends BaseComponent<INavProps, INavState> implements INav
       return true;
     }
 
-    if (location.protocol + '//' + location.host + location.pathname === target) {
-      return true;
-    }
-
-    if (location.hash) {
-      // Match the hash to the url.
-      if (location.hash === link.url) {
+    // If selectedKey is not defined in state, then check URL to determine link selected status
+    if (!this.state.selectedKey) {
```

Comment:
Line 297 of this file (same method) is already checking `state.selectedKey`. Would be nice to move this logic up there and always return early if either `props.selectedKey` or `state.selectedKey` is set.